### PR TITLE
boost: add support for python >=3

### PR DESCRIPTION
**Pull request by svenpeter42**
boost needs to be compiled independently for python 2 and 3. this is required for the WIP aiming to add python 3 support to vigra.

### CODE REVIEW

File: boost/meta.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ source:
 build:
   number: 5
   detect_binary_files_with_prefix: true
+  string: py{{CONDA_PY}}
```
Comment (stuarteberg) [inline boost/meta.yaml]: Please include the build number in the build string:

``` yaml
build:
  {% set build_num = 5 %}
  number: {{build_num}}
  string: py{{CONDA_PY}}_{{build_num}}
```

When conda-build 1.18.3 is finally released, the following syntax will work:

``` yaml
build:
  number: 5
  string: py{{CONDA_PY}}_{{PKG_BUILDNUM}}
```

**Comment by stuarteberg** [comment]
Nice PR.  I have a separate branch somewhere that deals with this problem via symlinks to the names that boost is expecting, but I like your solution more.

**Comment by svenpeter42** [comment]
There's still a small issue with this configuration: How do we make sure that the "correct" boost build is selected for e.g. vigra? 
If I just use this branch and then compile boost with py35 and py27 the py35 seems to always be used because vigra just depends on "boost 1.55.0". The official anaconda channel seems to solve this by adding explicit build strings to the dependencies, see e.g. https://repo.continuum.io/pkgs/free/linux-64/repodata.json, but that seems like a hack to me.

**Comment by stuarteberg** [comment]
> How do we make sure that the "correct" boost build is selected for e.g. vigra? 

We should specify constraints on the python version in the recipe, and provide an appropriate build string.  Since we've only used python 2.7 until now, I'm guessing several recipes in `ilastik-build-conda` will need to be fixed.

Something like this should work:

``` yaml
requirements:
  build:
    - python 2.7*|3.5*
    - python {{PY_VER}}*
  run:
    - python {{PY_VER}}*
```

I would have written a `build` spec of `(2.7*|3.5*),{{PY_VER}}*`, but parentheses aren't allowed in the conda [match spec language](http://conda.pydata.org/docs/spec.html#package-match-specifications).  Fortunately, the same effect can be achieved by simply adding an additional constraint on the next line.  The above requirements say "This package must be built with the version of python the user selected from the command line, which must be either `2.7` or `3.5`.  Also, it can only be run with the exact version the user chose."

You should add those lines to vigra's `meta.yaml`, too.

**Comment by k-dominik** [comment]
closing this, as far as I see it, python 3 support for vigra has been takled in the vigra-feedstock

**Pull request closed without merging**